### DISCUSSION TITLE
Want to create a "Child record", from a "Parent"

### DISCUSSION THREAD

**Discussion by mjashanks** [Budibase/budibase#938] [Help] [answered]
I wanted to make a button called "Add Note", to a "Signup Detail" screen (a Signup is just a person). This button should create a new note, with a relationship to the Contact, using a textbox in the same form.

This:

![image](https://user-images.githubusercontent.com/3524181/102512826-61764880-4082-11eb-93e6-7e2761c79e95.png)

However, there is no way to create the relationship. I think the reason for this is because relationship is an array.

![image](https://user-images.githubusercontent.com/3524181/102514845-e5313480-4084-11eb-9c54-7a2a568cd54f.png)

In my experience, this is a pretty common use case. what would probably be useful here, and in other places, would be to allow relationships to be One-Many - e.g. One Signup can have Many notes.

**Comment by sergey-dme** [comment, marked as answer]
Is there a way in the latest version to do this? I tried almost everything, changing key fields in parent table, used different field types, used `_id`  but nothing works.
The possibility to create a child record from a parent is a core of functionality in my app - I have "Discrepancy reports" table and want to create a child record - "Action taken" which has date/time, Action description and Result fields. One "Discrepancy report" can have many "Action taken" records.
Please help 🙏

**Reply by shogunpurple** [reply]
@sergey-dme this is possible with a little bit of handlebars.

Here's a contrived example that should help with some steps:
- You need to set up another form alongside the Edit form for your `Discrepancy Report`

![Screenshot 2021-06-30 at 10 29 25](https://user-images.githubusercontent.com/11256663/123937415-0c4eb400-d98e-11eb-9031-e4d5d70f4f1f.png)

- Update the "Save" actions in the Action taken form to look something like the following. The main thing to realise here is the use of `arrayify`, which will set up an array for the relationship.
![Screenshot 2021-06-30 at 10 29 56](https://user-images.githubusercontent.com/11256663/123937481-1f618400-d98e-11eb-9d2f-086c9e69b192.png)

- When you click the Create button in that form, it will create actions underneath that specific Discrepancy report.
![Screenshot 2021-06-30 at 10 33 33](https://user-images.githubusercontent.com/11256663/123938017-a31b7080-d98e-11eb-9620-3e842a1d2937.png)

Here's the exported version of the app above also - might be useful for you to import this into your budibase installation and see how it is set up.

[export(6).txt](https://github.com/Budibase/budibase/files/6739733/export.6.txt)

Hope this helps.

**Reply by sergey-dme** [reply]
@shogunpurple Thanks a lot for this! `arrayify` is the solution I was looking for. Thank you again for your time, appreciate it!

**Reply by hendrikjap** [reply]
This should be put in the docs or tutorial! It's very useful.
